traitsui/qt4/extra/progress_renderer.py

```diff
@@ -0,0 +1,52 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2016, Enthought, Inc.
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+#
+# Author: Corran Webster
+#------------------------------------------------------------------------------
+
+""" A renderer which displays a progress bar based on a 0.0 to 1.0 value.
+"""
+
+# System library imports
+from pyface.qt import QtCore, QtGui
+
+# ETS imports
+from traitsui.qt4.table_editor import TableDelegate
+
+
+class ProgressRenderer(TableDelegate):
+    """ A renderer which displays a progress bar for a True value and an
+        unchecked box for a false value.
+    """
+
+    #---------------------------------------------------------------------------
+    #  QAbstractItemDelegate interface
+    #---------------------------------------------------------------------------
+
+    def paint(self, painter, option, index):
+        """ Reimplemented to paint the checkbox.
+        """
+        # Get the column and object
```

Comment:
Not needed, I believe - this is basically a Python version of the code here: http://doc.qt.io/qt-4.8/qabstractitemdelegate.html#details

The style's `drawControl` should handle everything.